### PR TITLE
[17509] Feature: Ignore empty API

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
## Description

This PR introduces the required API to implement the ignore DDS entity feature.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [N/A] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [NO] Changes are ABI compatible. 
- [NO] Changes are API compatible. 
- [x] New feature has been added to the `versions.md` file (if applicable).
- [N/A] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [N/A] Applicable backports have been included in the description.

As this feature is not yet implemented, it will be documented in a follow-up PR when the implementation is made.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
Uncrustify fails because it tries to remove a required `;` from an `enum` declaration.

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test mac

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
Running the test suite because one mock has been modified in order to fix a Mac warning.

@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
Test failure does not seem related.

File: include/fastdds/dds/domain/DomainParticipantListener.hpp
Comment (MRicoIE2CS) [inline include/fastdds/dds/domain/DomainParticipantListener.hpp:1]: Please, add `[in]` and `[out]` marks on the Doxygen parameters for any new/modified methods.

File: include/fastdds/rtps/builtin/discovery/participant/PDP.h
Comment (MRicoIE2CS) [inline include/fastdds/rtps/builtin/discovery/participant/PDP.h:1]: Please, add `[in]` and `[out]` marks on the Doxygen parameters for any new/modified methods.

File: include/fastdds/rtps/participant/RTPSParticipant.h
Comment (MRicoIE2CS) [inline include/fastdds/rtps/participant/RTPSParticipant.h:1]: Please, add `[in]` and `[out]` marks on the Doxygen parameters for any new/modified methods.

File: include/fastdds/rtps/participant/RTPSParticipantListener.h
Comment (MRicoIE2CS) [inline include/fastdds/rtps/participant/RTPSParticipantListener.h:1]: Please, add `[in]` and `[out]` marks on the Doxygen parameters for any new/modified methods.

File: src/cpp/rtps/participant/RTPSParticipantImpl.h
Comment (MRicoIE2CS) [inline src/cpp/rtps/participant/RTPSParticipantImpl.h:1]: Please, add `[in]` and `[out]` marks on the Doxygen parameters for any new/modified methods.

**Review by MRicoIE2CS — CHANGES_REQUESTED** [review]

**Review by MRicoIE2CS — CHANGES_REQUESTED** [review]
There are Uncrustify issues.

**Comment by JLBuenoLopez-eProsima** [comment]
> There are Uncrustify issues.

Please, read [my comment](https://github.com/eProsima/Fast-DDS/pull/3354#issuecomment-1463589417) above about this issue.

File: include/fastdds/dds/domain/DomainParticipantListener.hpp
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/dds/domain/DomainParticipantListener.hpp:1]: Not really required, but I agree that it is nice to have

File: include/fastdds/dds/domain/DomainParticipantListener.hpp [hunk 1]
```diff
@@ -104,7 +124,26 @@ class DomainParticipantListener :
...
+
+    /*!
+     * This method is called when a new Subscriber is discovered, or a previously discovered subscriber changes
+     * its QOS or is removed.
+     *
+     * @param participant Pointer to the Participant which discovered the remote subscriber.
+     * @param info Remote subscriber information. User can take ownership of the object.
+     * @param should_be_ignored Flag to indicate the library to automatically ignore the discovered Participant.
+     */
+    virtual void on_subscriber_discovery(
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/dds/domain/DomainParticipantListener.hpp:139]: I think it is!

File: include/fastdds/dds/domain/DomainParticipantListener.hpp [hunk 2]
```diff
@@ -118,7 +157,26 @@ class DomainParticipantListener :
...
+
+    /*!
+     * This method is called when a new Publisher is discovered, or a previously discovered publisher changes
+     * its QOS or is removed.
+     *
+     * @param participant Pointer to the Participant which discovered the remote publisher.
+     * @param info Remote publisher information. User can take ownership of the object.
+     * @param should_be_ignored Flag to indicate the library to automatically ignore the discovered Participant.
+     */
+    virtual void on_publisher_discovery(
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/dds/domain/DomainParticipantListener.hpp:172]: This one too!

**Review by MRicoIE2CS — APPROVED** [review]
LGTM

**Review by MRicoIE2CS — APPROVED** [review]
LGTM

**Pull request merged**